feat(packageRules): add merge confidence matcher (#21049)
renovatebot/renovate: docs/usage/configuration-options.md

}
```

### matchConfidence

<!-- prettier-ignore -->
!!! warning

> This configuration option is dependent on an API key, therefore it is effectively disabled if none is set.

Reviewer: ```suggestion
This configuration option is dependent on a Mend API key, and is in private beta testing only. API keys are not available for free or via this repository.
```

Reviewer: @viceice like this?

Reviewer 2: yes

> !!! warning
> This configuration option is dependent on a Mend API key, and is in private beta testing only. API keys are not available for free or via this repository.

Reviewer 3: ```suggestion
!!! warning
    This configuration option needs a Mend API key, and is in private beta testing only.
    API keys are not available for free or via the `renovatebot/renovate` repository.
```

The formatting was nearly right, you need 4 spaces on each line. 😉 While I'm at it, a small rewrite to clarify.

For example to group high merge confidence updates:

```json
{
  "packageRules": [
    {
      "matchConfidence": ["high", "very high"],
      "groupName": "high merge confidence"
    }
  ]
}
```

> This feature requires an API token. Set it via Host rules under the `merge-confidence` HostType:

Reviewer 2: we shouldn't say twice, the warning above already mentioned the token requirement.

we should also add a warning that this is currently in private beta testing, so we don't get requests how to get that token.

> This feature requires an API token. Set it via Host rules under the `merge-confidence` HostType:

Reviewer: ```suggestion
Tokens can be configured via `hostRules` using the `merge-confidence` `hostType`:
```

> Tokens can be configured via `hostRules` using the `merge-confidence` `hostType`:

Reviewer 3: ```suggestion
Tokens can be configured via `hostRules` using the `"merge-confidence"` `hostType`:
```

I think we need to match the `"` in the JSON example here?
